src/query-builder/SelectQueryBuilder.ts

```
@@ -650,13 +650,13 @@ export class SelectQueryBuilder<Entity extends ObjectLiteral>
      * You also need to specify an alias of the joined data.
      * Optionally, you can add condition and parameters used in condition.
      */
-    innerJoinAndMapMany(
-        mapToProperty: string,
-        entity: Function | string,
+    innerJoinAndMapMany<Key extends string, Joined = unknown>(
+        mapToProperty: Key | `${string}.${Key}`,
+        entity: { new (): Joined } | Function | string,
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)